driver-core/src/main/java/com/datastax/driver/core/ScyllaCloudDatacenter.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (C) 2022 ScyllaDB
+ */
+package com.datastax.driver.core;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.net.HostAndPort;
+import com.google.common.net.InternetDomainName;
+import java.io.File;
+import java.net.InetSocketAddress;
+import java.net.MalformedURLException;
+import java.net.URL;
+
+class ScyllaCloudDatacenter {
+  private final String certificateAuthorityPath;
+  private final byte[] certificateAuthorityData;
+  private final String server;
+  private final String TLSServerName;
+  private final String nodeDomain;
+  private final String proxyURL;
+
+  @JsonCreator
+  public ScyllaCloudDatacenter(
+      @JsonProperty(value = "certificateAuthorityPath") String certificateAuthorityPath,
+      @JsonProperty(value = "certificateAuthorityData") byte[] certificateAuthorityData,
+      @JsonProperty(value = "server") String server,
+      @JsonProperty(value = "tlsServerName") String TLSServerName,
+      @JsonProperty(value = "nodeDomain") String nodeDomain,
+      @JsonProperty(value = "proxyURL") String proxyURL) {
+    this.certificateAuthorityPath = certificateAuthorityPath;
+    this.certificateAuthorityData = certificateAuthorityData;
+    this.server = server;
+    this.TLSServerName = TLSServerName;
+    this.nodeDomain = nodeDomain;
+    this.proxyURL = proxyURL;
+  }
+
+  public void validate() {
+    if (certificateAuthorityData == null) {
+      if (certificateAuthorityPath == null) {
+        throw new IllegalArgumentException(
+            "Either certificateAuthorityData or certificateAuthorityPath must be provided for datacenter description.");
+      }
+      File file = new File(certificateAuthorityPath);
+      if (!file.canRead()) {
+        throw new IllegalArgumentException(
+            "Cannot read file at given certificateAuthorityPath ("
+                + certificateAuthorityPath
+                + ").");
+      }
+    }
+    if (server == null) {
+      throw new IllegalArgumentException("server property is required in datacenter description.");
+    } else {
+      try {
+        // Property 'server' is not a true URL because it does not contain protocol prefix
+        // We're adding prefix just to satisfy that part of validation
+        URL url = new URL("http://" + server);
+        if (url.getPort() == -1) {
+          throw new IllegalArgumentException(
+              "server property '" + server + "' does not contain a port.");
+        }
+      } catch (MalformedURLException e) {
+        throw new IllegalArgumentException(
+            "server property '" + server + "' is not a valid URL", e);
+      }
+    }
+    if (nodeDomain == null || nodeDomain.length() == 0) {
+      throw new IllegalArgumentException(
+          "nodeDomain property is required in datacenter description.");
+    } else {
+      if (nodeDomain.contains(" "))
```

Comment:
According to RFC 1123 a lowercase subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character. These condintions aren't checking it

Reply:
It seems i mixed up 'domain names' with 'hostnames' which have different rules. Will fix that.